### PR TITLE
Rewritten and commented src/logs.py

### PR DESCRIPTION
**Pull request by nanometer5088**
The logging logic was confusing and not commented, so I've parcially rewritten the code and throughly commented it.

### CODE REVIEW

**Comment by nanometer5088** [comment]
Just adding that it does not change any code functionality - the log will work the same way as it did before.

It's just cleaner and easier to contribute now :1st_place_medal:

File: src/logs.py [hunk 1]
```diff
@@ -1,28 +1,42 @@
+# This code imports the `glob`, `os`, and `time` modules
```
Comment (zayKenyon) [inline src/logs.py:1]: Please do not use descriptive comments for lines like this, it's obvious what is done here.

File: src/logs.py [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
 import glob
 import os
 import time
 
+# The `Logging` class is defined
```
Comment (zayKenyon) [inline src/logs.py:6]: ^

File: src/logs.py [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
 class Logging:
+    # The class constructor initializes the `logFileOpened` attribute as `False`
```
Comment (zayKenyon) [inline src/logs.py:8]: ^
A better comment would be to explain what that attribute does, but that's tenuous.

File: src/logs.py [hunk 4]
```diff
@@ -1,28 +1,42 @@
...
-        if self.logFileOpened:
-            with open(f"logs/log-{max(filenames)}.txt", "a") as logFile:
-                logFile.write(f"[{time.strftime('%Y.%m.%d-%H.%M.%S', time.localtime(time.time()))}]"
-                              f" {stringToLog.encode('ascii', 'replace').decode()}\n")
-        else:
-            with open(f"logs/log-{max(filenames) + 1}.txt", "w") as logFile:
-                self.logFileOpened = True
-                logFile.write(f"[{time.strftime('%Y.%m.%d-%H.%M.%S', time.localtime(time.time()))}]"
-                              f" {stringToLog.encode('ascii', 'replace').decode()}\n")
+    # The `log` method takes a string `log_string` as an argument
```
Comment (zayKenyon) [inline src/logs.py:12]: ^
A better comment would be to explain what `log_string` is.

File: src/logs.py [hunk 5]
```diff
@@ -1,28 +1,42 @@
...
+    def log(self, log_string: str):
+        # The logs directory is determined
```
Comment (zayKenyon) [inline src/logs.py:14]: ^

File: src/logs.py [hunk 6]
```diff
@@ -1,28 +1,42 @@
...
+        logs_directory = os.getcwd() + "/logs"
+
+        # If the logs directory does not exist, it is created
```
Comment (zayKenyon) [inline src/logs.py:17]: I'd argue this is unnecessary, `if not os.path.exists()` reads similarly to standard English.

File: src/logs.py [hunk 7]
```diff
@@ -1,28 +1,42 @@
...
+        if not os.path.exists(logs_directory):
+            os.mkdir(logs_directory)
+        
+        # A list of log files is obtained using the `glob` module
```
Comment (zayKenyon) [inline src/logs.py:21]: ^

File: src/logs.py [hunk 8]
```diff
@@ -1,28 +1,42 @@
...
+        log_files = glob.glob(r"logs/log-*.txt")
+
+        # The log file numbers are extracted from the filenames
```
Comment (zayKenyon) [inline src/logs.py:24]: This is a good comment.

File: src/logs.py [hunk 9]
```diff
@@ -1,28 +1,42 @@
...
+        log_file_numbers = [int(file[9:-4]) for file in log_files]
+        
+        # If no log files exist, a log-0.txt file is created
```
Comment (zayKenyon) [inline src/logs.py:27]:
```suggestion
        # Creates a log file if none are present.
```

File: src/logs.py [hunk 10]
```diff
@@ -1,28 +1,42 @@
...
+        if not log_file_numbers:
+            log_file_numbers.append(0)
+        
+        # The log file name is determined
```
Comment (zayKenyon) [inline src/logs.py:31]: A better comment would break down the different parts of the statement.

File: src/logs.py [hunk 11]
```diff
@@ -1,28 +1,42 @@
...
+        log_file_name = f"logs/log-{max(log_file_numbers) + 1 if not self.logFileOpened else max(log_file_numbers)}.txt"
+
+        # The log file is opened and written to
```
Comment (zayKenyon) [inline src/logs.py:34]: Unnecessary.

File: src/logs.py [hunk 12]
```diff
@@ -1,28 +1,42 @@
...
+        with open(log_file_name, "a" if self.logFileOpened else "w") as log_file:
+            self.logFileOpened = True
+
+            # The current time is formatted
```
Comment (zayKenyon) [inline src/logs.py:38]: ^

File: src/logs.py [hunk 13]
```diff
@@ -1,28 +1,42 @@
...
+            current_time = time.strftime("%Y.%m.%d-%H.%M.%S", time.localtime(time.time()))
+
+            # The log string is written to the file with the current time as a prefix
```
Comment (zayKenyon) [inline src/logs.py:41]: ^

**Review by zayKenyon — CHANGES_REQUESTED** [review]

**Comment by zayKenyon** [comment]
In general I'd say to make sure you're not stating the obvious. Common philosophy is that comments should be used when code is hard to read; when reading good code, you won't need comments. I appreciate your syntax and grammar within the comments though.

**Comment by nanometer5088** [comment]
I've changed what you suggested, take a look :man_technologist:

File: src/logs.py [hunk 14]
```diff
@@ -1,20 +1,17 @@
...
-# The `Logging` class is defined
 class Logging:
-    # The class constructor initializes the `logFileOpened` attribute as `False`
+    # logFileOpened is a variable that keeps track of the log file status. 
+    # It is initialized as False to represent the log file wasn't open.
     def __init__(self):
         self.logFileOpened = False
 
-    # The `log` method takes a string `log_string` as an argument
+    # log_string is a string that represents the log message that will be written to the log file.
```
Comment (zayKenyon) [inline src/logs.py:11]:
```suggestion
    # `log_string` is a string that is the log message that will be written to the log file. 
```

**Review by zayKenyon — CHANGES_REQUESTED** [review]

File: src/logs.py (hunk 14 above)
Comment (nanometer5088) [reply src/logs.py:11]: Ah, fair enough.

File: src/logs.py (hunk 7 above)
Comment (zayKenyon) [reply src/logs.py:18]: Please address this.

Comment (nanometer5088) [reply src/logs.py]: done

**Review by zayKenyon — APPROVED** [review]

**Comment by nanometer5088** [comment]
Oh damn wrong button sorry

**Review by zayKenyon — APPROVED** [review]

**Pull request merged**